internal/handlers/common/filter.go

```diff
@@ -458,3 +465,62 @@ func filterFieldExprBitsAnySet(fieldValue, maskValue any) (bool, error) {
 
 	return true, nil
 }
+
+// filterFieldMod handles {field: {$mod: [divisor, remainder]}}
+func filterFieldMod(fieldValue, exprValue any) (bool, error) {
+	//	fmt.Printf("%+v\t%+v\n", fieldValue, expr)
+	arr := exprValue.(*types.Array)
+	if arr.Len() < 2 {
+		return false, NewErrorMsg(ErrBadValue, "$mod, not enough elements")
+	}
+	if arr.Len() > 2 {
+		return false, NewErrorMsg(ErrBadValue, "$mod, too many elements")
+	}
+
+	div, err := toInt64(must.NotFail(arr.Get(0)))
+	if err != nil {
+		return false, err
+	}
+	rem, err := toInt64(must.NotFail(arr.Get(1)))
+	if err != nil {
+		return false, err
+	}
+	field, err := toInt64(fieldValue)
+	if err != nil {
+		return false, err
+	}
+
+	if field%div != rem {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func toInt64(val any) (int64, error) {
+
+	switch v := val.(type) {
+	case float64:
+		return int64(v), nil
+	case float32:
```

Comment:
Most of those types are not possible; only 3 numerics types are allowed: float64, int32, int64. See there: https://github.com/FerretDB/FerretDB/blob/main/internal/types/types.go